modules/ximgproc/src/thinning.cpp

```diff
@@ -0,0 +1,67 @@
+#include "precomp.hpp"
+
+using namespace std;
+
+namespace cv {
+namespace ximgproc {
+
+// Function prototypes to suppress -Wmissing-declarations warnings
+void thinningIteration(Mat img, int iter);
+void thinning(InputArray _input, OutputArray _output);
+
+// Applies a thinning iteration to a binary image
+void thinningIteration(Mat img, int iter){
```

Comment:
Well, additional include is not required. It is done in [precomp.hpp file](https://github.com/opencv/opencv_contrib/blob/master/modules/ximgproc/src/precomp.hpp)